detect_range_and_count_readings.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Dict, Tuple
+
+
+def reset_values(inputs, index):
+    start = inputs[index]
+    end = inputs[index]
+    num = 1
+    return start, end, num
+
+
+def detect_range_and_count_readings(inputs: List[int]):
+    # Output variable to store list of ranges along with number of readings
+    output: List[Dict[Tuple[int, int], int]] = []
+
+    # Sort inputs
+    inputs.sort()
+
+    # Define placeholders for range identification and count
+    start_range, end_range, num = reset_values(inputs, 0)
+
+    for i in range(1, len(inputs)):
+        eval = (inputs[i] == end_range + 1) | (inputs[i] == end_range)
```

Comment:
Using logical OR instead of bitwise OR id preferred in logical comparisons. Bitwise OR is not same as logical OR.
It might work in this case, but generally at some other code , it should be carefully selected for use.

Reply:
Thank you s for your comments. 

In this case, the use of Bitwise OR was intentional to reduce the CCN to 3 as by using Logical OR a new execution gets created while its not the case with Bitwise OR.

Reply:
👍